tests/aggregation/tests.py

```diff
@@ -1024,7 +1033,11 @@ def be_evil(self, compiler, connection):
             b1 = qs.get(pk=self.b4.pk)
             self.assertEqual(b1.sums, 2)
         finally:
-            delattr(Sum, 'as_' + connection.vendor)
+            if as_existing is None:
+                delattr(Sum, 'as_' + connection.vendor)
+            else:
+                # restore the method
+                setattr(Sum, 'as_' + connection.vendor, as_existing)
```

Comment:
Wouldn't it be simpler (and friendlier to parallelization), instead of monkey-patching the canonical `Sum`, to start this method with `class MySum(Sum): pass` and then use `MySum` instead? We could even define the inner functions as proper methods then.